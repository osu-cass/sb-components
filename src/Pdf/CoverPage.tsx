import * as React from 'react';

export interface CoverPageProps {
    subject: string;
    grade: string;
    date?: {
        getMonth: () => string;
        getDate: ()  => string;
        getFullYear:  () => string;
    }
}

export class CoverPage extends React.Component<CoverPageProps, {}> {
    render() {
        // the data prop is for snapshot testing
        const today = this.props.date ? this.props.date : new Date();
        return (
            <div className="page first-page">
                <div className="first-page-title">
                    Smarter Balanced<br />
                    Assessment Consortium
                </div>
                <div className="first-page-subtitle">
                    {this.props.subject} Practice Test Scoring Guide<br />
                    <strong>{this.props.grade}</strong>
                </div>
                <div className="first-page-date">
                    {today.getMonth()}/{today.getDate()}/{today.getFullYear()}
                </div>
            </div>
        );
    }
}