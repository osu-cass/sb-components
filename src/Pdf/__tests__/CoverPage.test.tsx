import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { CoverPage } from '../CoverPage';
import { shallow, mount, render } from 'enzyme';

const date = {
    getMonth: () => "01",
    getDate: ()  => "12",
    getFullYear:  () => "2017"
}

describe("First page", () => {
    it("matches snapshot", () => {
        let wrapper = shallow(<CoverPage subject="math" grade="grade 3" date={date}/>);
        expect(wrapper).toMatchSnapshot();
    })
})