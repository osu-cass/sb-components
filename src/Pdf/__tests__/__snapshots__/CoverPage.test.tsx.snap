// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First page matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CoverPage
    date={
      Object {
        "getDate": [Function],
        "getFullYear": [Function],
        "getMonth": [Function],
      }
    }
    grade="grade 3"
    subject="math"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="first-page-title"
        >
          Smarter Balanced
          <br />
          Assessment Consortium
        </div>,
        <div
          className="first-page-subtitle"
        >
          math
           Practice Test Scoring Guide
          <br />
          <strong>
            grade 3
          </strong>
        </div>,
        <div
          className="first-page-date"
        >
          01
          /
          12
          /
          2017
        </div>,
      ],
      "className": "page first-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Smarter Balanced",
            <br />,
            "Assessment Consortium",
          ],
          "className": "first-page-title",
        },
        "ref": null,
        "rendered": Array [
          "Smarter Balanced",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Assessment Consortium",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "math",
            " Practice Test Scoring Guide",
            <br />,
            <strong>
              grade 3
            </strong>,
          ],
          "className": "first-page-subtitle",
        },
        "ref": null,
        "rendered": Array [
          "math",
          " Practice Test Scoring Guide",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "grade 3",
            },
            "ref": null,
            "rendered": "grade 3",
            "type": "strong",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "01",
            "/",
            "12",
            "/",
            "2017",
          ],
          "className": "first-page-date",
        },
        "ref": null,
        "rendered": Array [
          "01",
          "/",
          "12",
          "/",
          "2017",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="first-page-title"
          >
            Smarter Balanced
            <br />
            Assessment Consortium
          </div>,
          <div
            className="first-page-subtitle"
          >
            math
             Practice Test Scoring Guide
            <br />
            <strong>
              grade 3
            </strong>
          </div>,
          <div
            className="first-page-date"
          >
            01
            /
            12
            /
            2017
          </div>,
        ],
        "className": "page first-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Smarter Balanced",
              <br />,
              "Assessment Consortium",
            ],
            "className": "first-page-title",
          },
          "ref": null,
          "rendered": Array [
            "Smarter Balanced",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Assessment Consortium",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "math",
              " Practice Test Scoring Guide",
              <br />,
              <strong>
                grade 3
              </strong>,
            ],
            "className": "first-page-subtitle",
          },
          "ref": null,
          "rendered": Array [
            "math",
            " Practice Test Scoring Guide",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "grade 3",
              },
              "ref": null,
              "rendered": "grade 3",
              "type": "strong",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "01",
              "/",
              "12",
              "/",
              "2017",
            ],
            "className": "first-page-date",
          },
          "ref": null,
          "rendered": Array [
            "01",
            "/",
            "12",
            "/",
            "2017",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
